### PR TITLE
[Feature-9772][plugin/ui] support SparkSQL Task

### LINKED ISSUE #9772: [Feature][plugin/ui] support spark sql for SparkTask
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

## What happened
According to user requirements, we hope to support spark sql for SparkTask.

## What you expected to happen
Users want to have an editor to fill in the sql script, which is convenient for ETL.

## Brief change log

- ProgramType.class add sql type
- SparkTask.class and SparkArgsUtils.class add main logical for support Spark sql command
- SparkTask.class  add some logical for support spark sql command
- use-spark.ts adds some logic to support front-end pages supporting spark sql editing SQL

## The effect of the execution

Our desired front-end effect：
lQLPDhtd5ypCZK3NA5bNBCywAi-2JJ5_quwCaoYbSoA-AA_1068_918

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sq-q**
## Purpose of the pull request

According to user requirements, we hope to support spark sql for SparkTask.

Users want to have an editor to fill in the sql script, which is convenient for ETL.

The renderings we need are as follows:
lQLPDhtd5ypCZK3NA5bNBCywAi-2JJ5_quwCaoYbSoA-AA_1068_918

close #9772 

## Brief change log

- ProgramType.class add sql type
- SparkTask.class and SparkArgsUtils.class add main logical for support Spark sql command
- use-spark.ts adds some logic to support front-end pages supporting spark sql editing SQL

## Verify this pull request

This change added tests and can be verified as follows:

org.apache.dolphinscheduler.plugin.task.spark.SparkTaskTest

### CODE REVIEW

**Comment by labbomb** [comment]
@Amy0104 @devosend Take the time to review the front-end code

**Comment by songjianet** [comment]
Does this PR have a corresponding issue? @sq-q

**Comment by sq-q** [comment]
> Does this PR have a corresponding issue? @sq-q

This is my issue (https://github.com/apache/dolphinscheduler/issues/9772).

File: dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java
Comment (SbloodyS) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: I think we should print the info log in the task log instead of the worker log to facilitate user troubleshooting.

Comment (sq-q) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: Some code smells in sonarcloud are inappropriate, I am refactoring here, please wait a moment.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java
Comment (SbloodyS) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: Maybe we should reuse SparkTask's logger instead of redefining one. So that we can unify the log output.

Comment (sq-q) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: SparkArgsUtils and AbstractTaskExecutor have no inheritance relationship. If you use the same logger, you need to move SparkArgsUtils to SparkTask, and you can define only one logger. Do you think this method is OK?

Comment (SbloodyS) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: I'm not quite sure about that. @ruanwenjun @zhuangchong Do you have time to take a look at this PR?

Comment (sq-q) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: I have refactored.

Comment (SbloodyS) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: @sq-q Don't get me wrong. I'm just not familiar about this part. Please wait patiently for others to reply.

Comment (sq-q) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-spark/src/main/java/org/apache/dolphinscheduler/plugin/task/spark/SparkArgsUtils.java]: No, I think this method is also feasible, so I will change it first. If I find it inappropriate in the future, I will change it back.

**Review by zhuangchong — APPROVED** [review]
+1

**Review by SbloodyS — APPROVED** [review]
+1

**Comment by SbloodyS** [comment]
Thanks for your contribution. This is a good beginning. Considering that it's your first contribution, I think we can get deep communication, you can contact me through mail or add WeChat(SbloodyS), when mail or added, please tell me who you are. I think I can help to familiar with the Dolphinscheduler if you meet with problems. @sq-q

**Comment by sq-q** [comment]
> 感谢您的贡献。这是一个好的开始。考虑到是你的第一次投稿，我想我们可以深入交流，你可以通过邮件联系我或者加微信（SbloodyS），邮件或者加的时候请告诉我你是谁。如果您遇到问题，我想我可以帮助您熟悉 Dolphinscheduler。@sq-q

Grazie

**Comment by songjianet** [comment]
This kind of code that involves changes to the front and back ends should be noted that it needs to wait for the approval of the front end before it can be merged. @SbloodyS

**Comment by SbloodyS** [comment]
> This kind of code that involves changes to the front and back ends should be noted that it needs to wait for the approval of the front end before it can be merged. @SbloodyS

Sorry. I'll pay attention next time.

**Pull request merged**
